manifests/init.pp

```diff
@@ -40,6 +40,10 @@
 #   The directory used to install the report processor to
 # @param use_client_tls_certs
 #   Enable client TLS authentication to foreman
+# @param fact_watcher_service
+#   Sets up a simple systemd unit that watches for new fact files and publishes them to foreman. Not required when foreman is the ENC
```

Comment:
There is a `$enc` parameter which you can use. Perhaps `fail()` if both are true?

Technically you could look at `$enc_upload_facts` but then you don't have a guarantee that the facts are uploaded before the ENC is calculated. Uploading facts will create a host entry when it doesn't exist, so I wouldn't do that.